lib/packwerk/application_load_paths.rb

```diff
@@ -10,18 +10,14 @@ class << self
 
       sig { returns(T::Array[String]) }
       def extract_relevant_paths
-        assert_application_booted
+        return [] unless defined?(::Rails)
```

Comment:
Is there a way to return `[]` at the end of this method instead of in an early return? 

`defined?(::Rails)` is really misleading because it (almost) always this will be true, as Packwerk itself has Rails and will fall back to that.

In the case where `defined?(::Rails)` works the way we intend it too though:

In the context of `ApplicationLoadPaths`, we would still want to raise if Rails is not defined because the point of the class is to get the Rails load paths. Returning an empty array would be misleading users to the actual functionality API of this class. 

TL;DR: We should not return empty array here but instead outside of this class, as it would be misleading to the functionality of this class.

Reply:
As per [my latest comment](https://github.com/Shopify/packwerk/pull/77#issuecomment-733284451), we can just leave this class as is and come back to add the Rails loading logic in another PR. When that is added, we won't have to raise when Rails is not defined, and just return an empty array as you proposed here. 

We could make the change now and return an empty array, but it would be hiding the fact that Rails _should_ be defined/loaded but is not. 

Unless someone has strong opinions, I say we keep the raise until we can make that change.

Reply:
Ok, thanks @wildmaples, implemented in fe1697e

If we want to avoid raising for non-rails apps in the meantime, then I could implement [the second suggestion](https://github.com/Shopify/packwerk/pull/77#issuecomment-733002795) from @rochlefebvre instead. But if the PR for the Rails logic is going to be worked on sooner rather than later, perhaps the raise isn't too much of a concern for now.